client/auth/auth-code-button.jsx

```diff
@@ -15,6 +12,9 @@ import Notice from 'components/notice';
 
 const initialState = { status: 'ready', errorLevel: false, errorMessage: false };
 
+// The tests need an absolute URL, so we need to detect if Jest is running.
+const SMS_URL = process.env.JEST_WORKER_ID !== undefined ? 'http://localhost/sms' : '/sms';
```

Comment:
Is there any way to avoid this? Where exactly is an absolute URL needed? In Nock?

Reply:
The issue is that nock uses the node `fetch` polyfill, and relative URLs don't work there since they're meaningless on the server side.

I found a workaround, though, by wrapping the global `fetch` in a function that transforms any relative URLs into absolute ones, with `http://localhost` as a base.

Thanks for poking me into fixing this the right way 🙂